syntax = "proto3";

package twitterdash;
import "google/protobuf/timestamp.proto";

enum TrendType{
    Topic = 0;
    Hashtag = 1;
}

enum Granularity{
    minute = 0;
    hour = 1;
    day = 2;
}

message Trend{
    TrendType trendType = 1;
    string name = 2;
    int32 country = 3; // WOEID of the country
    int32  placement = 4; // Place of the Tweet in the given Country
    optional int32 tweetVolume24= 5; // Number of Tweets in the given Country
}

message TrendProviderReply{
    google.protobuf.Timestamp timestamp = 1;
    repeated Trend trends = 2;
}

message Tweet{
    int64 ID = 1;
    int64 Conversation_ID = 2;
    google.protobuf.Timestamp timestamp = 3;
    string Text = 4;
    int64 UserID = 5;
    int32 likes = 6;
    int32 replies = 7;
    int32 retweets = 8;
    repeated string Hashtags = 9;
    string language = 10;
}